guacml/enums.py

```diff
@@ -1,6 +0,0 @@
-from enum import Enum
-
-ProblemType = Enum('ProblemType', 'BINARY_CLAS MULTI_CLAS REGRESSION')
```

Comment:
Couldn't we write a custom serialiser/deserialiser. This way we could keep the type save python enums and also during deserialisation give out nice error messages for the users that prefer the json way. The tradeoff for me is not even so much about cleaner code but more about having the better error messages for our users so that they don't run into errors just for misspelling the error metric they wanna use. WDYT?